datafusion/physical-expr/src/utils.rs

```diff
@@ -237,47 +173,105 @@ fn normalize_sort_requirement_with_equivalence_properties(
     sort_requirement
 }
 
-fn normalize_sort_requirement_with_ordering_equivalence_properties(
-    mut sort_requirement: PhysicalSortRequirement,
-    eq_properties: &[OrderingEquivalentClass],
-) -> PhysicalSortRequirement {
-    if let Some(options) = &mut sort_requirement.options {
-        (sort_requirement.expr, *options) =
-            normalize_expr_with_ordering_equivalence_properties(
-                sort_requirement.expr,
-                *options,
-                eq_properties,
-            );
+/// This function searches for the slice `section` inside the slice `given`.
+/// It returns each range where `section` is compatible with the corresponding
+/// slice in `given`.
+fn get_compatible_ranges(
+    given: &[PhysicalSortRequirement],
+    section: &[PhysicalSortRequirement],
+) -> Vec<Range<usize>> {
+    let n_section = section.len();
+    let n_end = if given.len() >= n_section {
+        given.len() - n_section + 1
+    } else {
+        0
+    };
+    (0..n_end)
+        .filter_map(|idx| {
+            let end = idx + n_section;
+            given[idx..end]
+                .iter()
+                .zip(section)
+                .all(|(req, given)| given.compatible(req))
+                .then_some(Range { start: idx, end })
+        })
+        .collect()
+}
+
+/// This function constructs a duplicate-free vector by filtering out duplicate
+/// entries inside the given vector `input`.
+fn collapse_vec<T: PartialEq>(input: Vec<T>) -> Vec<T> {
+    let mut output = vec![];
+    for item in input {
+        if !output.contains(&item) {
+            output.push(item);
+        }
     }
-    sort_requirement
+    output
 }
 
-pub fn normalize_sort_expr(
-    sort_expr: PhysicalSortExpr,
+pub fn normalize_sort_exprs(
```

Comment:
I wonder if we can add some doc comments here explaining what this code is doing (aka what is it normalizing against?)